### PR TITLE
Update cli and add cli integration test

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR updates the cli to the latest refactor first introduced in #129, along with adding an integration test for the cli.

TODO: 
- [x] switch back to the celestia-core branch `release/v0.34.x-celestia` after [#607](https://github.com/celestiaorg/celestia-core/pull/607) is merged
- [ ] maybe only run the integration test with the `integration` build tag, as it take a while

closes: #132 and #39

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -127,6 +127,6 @@ replace (
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
 	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v0.44.1-celestia
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
-	github.com/tendermint/tendermint v0.34.13 => github.com/celestiaorg/celestia-core v0.34.14-celestia.0.20211121093520-199a1e4232ac
+	github.com/tendermint/tendermint v0.34.13 => github.com/celestiaorg/celestia-core v0.34.14-celestia.0.20220105195031-3afbe43f198a
```
Comment (adlerjohn) [inline go.mod:130]: Is there a reason for this?

File: testutil/network/network.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+	net := network.New(t, config)
+
+	// add the keys to the keyring that is used by the integration test
+	for i, name := range genAccNames {
+		_, err := net.Validators[0].ClientCtx.Keyring.NewAccount(name, mnemonics[i], "", "", hd.Secp256k1)
+		require.NoError(t, err)
+	}
+
+	// t.Cleanup(net.Cleanup)
```
Comment (adlerjohn) [inline testutil/network/network.go:52]: What is this line for?

Comment (evan-forbes) [reply testutil/network/network.go]: it was adding a cleanup function to a test, but it ended being redundant. forgot to remove it, good catch!

https://github.com/celestiaorg/celestia-app/pull/171/commits/84baa00969501d419eb30e3e6410cb553a63fc06

File: go.mod (hunk 1 above)
Comment (evan-forbes) [reply go.mod]: yeah, we're updating to the latest commit of celestia-core `v0.34.x-celestia`, cause we need the event system to work in order for the integration test to work

**Review by Bidon15 — APPROVED** [review]
LGTM

**Pull request merged**